src/util/api.js

```diff
@@ -41,3 +41,67 @@ export async function resetPasswordRequest(token, newPassword, email) {
   if (!res.ok) throw new Error("Failed to reset password");
   return await res.json();
 }
+
+export async function getFilteredPosts(category, search) {
+  const params = new URLSearchParams();
+  if (category) params.append("category", category);
+  if (search) params.append("search", search);
+
+  const token = localStorage.getItem("token");
+
+  const res = await fetch(`${BASE_URL}/items?${params.toString()}`, {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${token}`,
+    },
+  });
+
+  const data = await res.json();
+  if (!res.ok) throw new Error(data.message || "Failed to fetch posts");
+
+  return data.items.map((item) => ({
+    ...item,
+    photos: item.images?.map((img) => img.image_url) || [],
+    category: item.category_name || item.Category?.category_name || "Other",
+    user: {
+      name: `${item.User?.first_name || ""} ${
+        item.User?.last_name || ""
+      }`.trim(),
+      email: item.User?.email || "",
+      avatar: item.User?.avatar_url || "",
+    },
+  }));
```

Comment:
This code is duplicated at `getPostById`. You can create a helper function that does the "normalization".

```js
function normalizeItem(item) {
  return {
    ...item,
    photos: item.images?.map((img) => img.image_url) || [],
    category: item.category_name || item.Category?.category_name || "Other",
    user: {
      name: `${item.User?.first_name || ""} ${
        item.User?.last_name || ""
      }`.trim(),
      email: item.User?.email || "",
      avatar: item.User?.avatar_url || "",
    },
  };
}
```

And could be used as...

```suggestion
  return data.items.map(normalizeItem);
```

Both here and at the `getPostById` function.

Reply:
Thanks, I extracted the normalization logic into a normalizeItem function to reuse it in both getFilteredPosts and getPostById